src/pwm.rs

```diff
@@ -258,10 +300,95 @@ macro_rules! hal {
                         .set_bit()
                 );
 
-                unsafe { mem::MaybeUninit::uninit().assume_init() }
+                Pwm {
+                    clk,
+                    _pins: PhantomData,
+                }
+            }
+
+            impl<REMAP, P, PINS> hal::Pwm for Pwm<$TIMX, REMAP, P, PINS> where
+                REMAP: Remap<Periph = $TIMX>,
+                PINS: Pins<REMAP, P>,
+            {
+                type Channel = Channel;
+                type Duty = u16;
+                type Time = Hertz;
+
+                fn enable(&mut self, channel: Self::Channel) {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 0) },
+                        Some(Channel::C2) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 4) },
+                        Some(Channel::C3) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 8) },
+                        Some(Channel::C4) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 12) },
+                        None => {}
+                    }
+                }
+
+                fn disable(&mut self, channel: Self::Channel) {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 0) },
+                        Some(Channel::C2) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 4) },
+                        Some(Channel::C3) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 8) },
+                        Some(Channel::C4) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 12) },
+                        None => {}
+                    }
+                }
+
+                fn get_duty(&self, channel: Self::Channel) -> Self::Duty {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { (*$TIMX::ptr()).ccr1.read().ccr().bits() },
+                        Some(Channel::C2) => unsafe { (*$TIMX::ptr()).ccr2.read().ccr().bits() },
+                        Some(Channel::C3) => unsafe { (*$TIMX::ptr()).ccr3.read().ccr().bits() },
+                        Some(Channel::C4) => unsafe { (*$TIMX::ptr()).ccr4.read().ccr().bits() },
+                        None => 0,
+                    }
+                }
+
+                fn set_duty(&mut self, channel: Self::Channel, duty: Self::Duty) {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { (*$TIMX::ptr()).ccr1.write(|w| w.ccr().bits(duty)) },
+                        Some(Channel::C2) => unsafe { (*$TIMX::ptr()).ccr2.write(|w| w.ccr().bits(duty)) },
+                        Some(Channel::C3) => unsafe { (*$TIMX::ptr()).ccr3.write(|w| w.ccr().bits(duty)) },
+                        Some(Channel::C4) => unsafe { (*$TIMX::ptr()).ccr4.write(|w| w.ccr().bits(duty)) },
+                        None => {},
```

Comment:
In this code:
```rust
pwm.disable(C1);
pwm.set_duty(C1, 0.5);
pwm.enable(C1);
```
I would expect it to get the updated duty after being enabled, but this seems to not be the case here. As far as I can tell, 

```rust
match channel {
    Channel::C1 => unsafe{ /*ccr1.write...*/ }
    Channel::C2 => unsafe{ /*ccr2.write...*/ }
    Channel::C3 => unsafe{ /*ccr3.write...*/ }
    Channel::C4 => unsafe{ /*ccr3.write...*/ }
}
```
would result in that behaviour